src/swap/SwapScreen.tsx

```diff
@@ -62,7 +63,126 @@ const DEFAULT_SWAP_AMOUNT: SwapAmount = {
   [Field.TO]: '',
 }
 
-type Props = NativeStackScreenProps<StackParamList, Screens.SwapScreenWithBack>
+interface SwapState {
+  fromTokenId: string | undefined
+  toTokenId: string | undefined
+  // Raw input values (can contain region specific decimal separators)
+  swapAmount: SwapAmount
+  updatedField: Field
+  selectingToken: Field | null
+  confirmingSwap: boolean
+  // Keep track of which swap is currently being executed from this screen
+  // This is because there could be multiple swaps happening at the same time
+  startedSwapId: string | null
+  switchedToNetworkId: NetworkId | null
+}
+
+function getInitialState(fromTokenId?: string): SwapState {
+  return {
+    fromTokenId,
+    toTokenId: undefined,
+    swapAmount: DEFAULT_SWAP_AMOUNT,
+    updatedField: Field.FROM,
+    selectingToken: null,
+    confirmingSwap: false,
+    startedSwapId: null,
+    switchedToNetworkId: null,
+  }
+}
+
+const swapSlice = createSlice({
+  name: 'swapSlice',
+  initialState: getInitialState,
+  reducers: {
+    changeAmount: (state, action: PayloadAction<{ field: Field; value: string }>) => {
+      const { field, value } = action.payload
+      state.confirmingSwap = false
+      state.startedSwapId = null
+      state.updatedField = field
+      if (!value) {
+        state.swapAmount = DEFAULT_SWAP_AMOUNT
+        return
+      }
+      // Regex to match only numbers and one decimal separator
+      const sanitizedValue = value.match(/^(?:\d+[.,]?\d*|[.,]\d*|[.,])$/)?.join('')
```

Comment:
do we really need to declare all these alternatives in regex?
probably we can simplify it a bit?
```suggestion
      const sanitizedValue = value.match(/^\d*[.,]?\d*$/)?.join('')
```

Reply:
Yes we may be able to simplify, since this isn't directly related to the focus of this PR I'll leave it as is.